### PR TITLE
Allow for the inclusion of static files

### PR DESCRIPTION
**Pull request by iiegn**
tackles #77

this is another - updated - shot at the issue (after i messed up pull request #78...). sorry for that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,15 @@
...
   - jekyll-sitemap
+
+# optionally, to include static files
+
+sitemap:
+  extensions:
+    - .html  # this is the default
+    - .htm   # this is the default
+    - .xhtml # this is the default
+    - .pdf
```
Comment (benbalter) [inline README.md]: 1. Why do we need the preceding `.`? Would there ever be an extension without one? can we normalize it and make it optional?

2. For the docs: Can you explain what this feature does and when I'd want to use it? If I want to add one additional extension, do I have to add all the defaults back in or is it cumulative? (if it's not, when would I *not* want to include the defaults?)

File: lib/jekyll-sitemap.rb [hunk 2]
```diff
@@ -15,7 +15,13 @@ class JekyllSitemap < Jekyll::Generator
     def generate(site)
       @site = site
       @site.config["time"]         = Time.new
-      @site.config["html_files"]   = html_files.map(&:to_liquid)
+      begin
+          exts = @site.config["sitemap"]["extensions"]
```
Comment (benbalter) [inline lib/jekyll-sitemap.rb]: Looks like this name spacing (which I agree with) will be a breaking change? Can we make it backwards compatible?

File: lib/jekyll-sitemap.rb [hunk 3]
```diff
@@ -15,7 +15,13 @@ class JekyllSitemap < Jekyll::Generator
...
+      rescue NameError
+          # safe default - and backward-compatible behaviour
+          exts = [".html", ".htm", ".xhtml"]
+      end
+      @site.config["sitemap_static_files"]  = sitemap_static_files(exts).map(&:to_liquid)
```
Comment (benbalter) [inline lib/jekyll-sitemap.rb]: Why are we storing this in the user-space site config?

File: lib/jekyll-sitemap.rb [hunk 4]
```diff
@@ -23,15 +29,9 @@ def generate(site)
...
-      .html
-      .xhtml
-      .htm
-    )
-
-    # Array of all non-jekyll site files with an HTML extension
-    def html_files
-      @site.static_files.select { |file| HTML_EXTENSIONS.include? file.extname }
+    # Array of all non-jekyll site files with given extensions
+    def sitemap_static_files (exts=[])
```
Comment (pathawks) [inline lib/jekyll-sitemap.rb]: Since this is in the `JekyllSitemap` class, the name feels redundant. Could we just call this `static_files` instead?

**Comment by pathawks** [comment]
@iiegn There are some conflicts that prevent me from being able to merge. Would you mind rebasing this PR?

**Comment by iiegn** [comment]
cleand-up, adapted to comments and rebased to PR. pls. take another look.

File: lib/jekyll-sitemap.rb [hunk 5]
```diff
@@ -29,9 +30,10 @@ def generate(site)
       .htm
     ).freeze
 
-    # Array of all non-jekyll site files with an HTML extension
-    def html_files
-      @site.static_files.select { |file| HTML_EXTENSIONS.include? file.extname }
+    # Array of all non-jekyll site files with an HTML and user defined file extensions
+    def nonsite_files
```
Comment (pathawks) [inline lib/jekyll-sitemap.rb]: I see what you were going for, but I don't like the name. Maybe `static_files`?

**Comment by pathawks** [comment]
The code is solid; everything is looking good.

I'll be honest, I was just about ready to merge this but then I read Ben's piece on [Optimizing for power users](http://ben.balter.com/2016/03/08/optimizing-for-power-users-and-edge-cases/).

I see the benefit of including PDF files in the sitemap. Instead of adding an option, could we just _always_ include PDF files in the sitemap? There is no reason to include images or other things in the sitemap; other than PDF I can't really think of any other file type a user would want to include.

**Comment by benbalter** [comment]
> I see the benefit of including PDF files in the sitemap. Instead of adding an option, could we just always include PDF files in the sitemap? There is no reason to include images or other things in the sitemap; other than PDF I can't really think of any other file type a user would want to include.

:sparkles: Came here to ask the same question. :smile:

**Comment by stephantabor** [comment]
> There is no reason to include images or other things in the sitemap; other than PDF I can't really think of any other file type a user would want to include.

Unless i'm misunderstanding something, I think that there are some people who would like to include video and images in sitemaps

* Video - https://support.google.com/webmasters/answer/80471?hl=en
* Images - https://support.google.com/webmasters/answer/178636?hl=en

**Comment by pathawks** [comment]
> Unless i'm misunderstanding something, I think that there are some people who would like to include video and images in sitemaps

A fair point, but video sitemaps require extra metadata which currently feels out of scope. What we are talking about here is just one more text-ish document type.

Thinking about how users may try to abuse the custom extension option to include images or video in the sitemap further convinces me that the Right Thing is to just add `.pdf` to the list of included files and call it good.

Adding support for videos and images involves much more than just whitelisting a file extension.

**Comment by iiegn** [comment]
personally, i started this because of ``.pdf`` files. so, that's that. 

however, while trying to get a clearer picture of the issue i did come across https://support.google.com/webmasters/answer/35287?hl=en, and there some extensions did catch my attention; notably, ``.tex, .od[pst], .txt, `` (note:google-centric. couldn't find out what e.g. bing indexes and whether a sitemap file helps...)

for what it's worth, once i set up the sitemap file the pdfs on my site have been getting indexed in batches (of course, before the sitemap file they also got indexed - they are linked after all - but in a less systematic fashion).

ultimately - following the [power user argument](http://ben.balter.com/2016/03/08/optimizing-for-power-users-and-edge-cases/) - its your decision ;)

**Comment by pathawks** [comment]
I have added PDF files to the sitemap in #109

Let us consider future additions on a case by case basis, instead of just making the user figure it out.

**Pull request closed without merging**